src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp

```diff
@@ -955,22 +1069,26 @@ namespace OpenMS
   {
     for (const OpenMS::Feature& f : fmap_input)
     {
-      if (f.metaValueExists("identifier"))
+      if (f.metaValueExists("PeptideRef") && f.metaValueExists("identifier"))
       {
-        auto found_f = fmapmap.emplace(f.getMetaValue("identifier").toStringList().at(0), std::vector<OpenMS::Feature>({f}));
+        std::string id = std::string(f.getMetaValue("PeptideRef")) + std::string("_") + std::string(f.getMetaValue("identifier").toStringList().at(0));
+        std::string id_f = id + std::string("_") + std::to_string(f.getRT());
+        auto found_f = fmapmap.emplace(id_f, std::vector<OpenMS::Feature>({f}));
         if (!found_f.second)
         {
-          fmapmap.at(f.getMetaValue("identifier").toStringList().at(0)).push_back(f);
+          fmapmap.at(id_f).push_back(f);
         }
       }
       for (const OpenMS::Feature& s : f.getSubordinates())
       {
-        if (s.metaValueExists("identifier"))
+        if (s.metaValueExists("PeptideRef") && s.metaValueExists("identifier"))
         {
-          auto found_s = fmapmap.emplace(s.getMetaValue("identifier").toStringList().at(0), std::vector<OpenMS::Feature>({s}));
+          std::string id = std::string(s.getMetaValue("PeptideRef")) + std::string("_") + std::string(s.getMetaValue("identifier").toStringList().at(0));
+          std::string id_s = id + std::string("_") + std::to_string(s.getRT());
+          auto found_s = fmapmap.emplace(id_s, std::vector<OpenMS::Feature>({s}));
           if (!found_s.second)
           {
-            fmapmap.at(s.getMetaValue("identifier").toStringList().at(0)).push_back(s);
+            fmapmap.at(id_s).push_back(s);
```

Comment:
looks identical to the case above. Use a Lambda?

Reply:
indeed. done.